sale_order_project/models/sale.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2014 Akretion - Sébastien BEAU <sebastien.beau@akretion.com>
+# Copyright 2014 Akretion - Benoît GUILLOT <benoit.guillot@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from datetime import date
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.multi
+    @api.depends('analytic_account_id')
+    def _compute_related_project_id(self):
+        for record in self:
+            project_domain = [(
+                'analytic_account_id', '=', record.analytic_account_id.id
+            )]
+            record.related_project_id = (
+                self.env['project.project'].search(project_domain, limit=1)[:1]
+            )
+
+    related_project_id = fields.Many2one(
+        comodel_name='project.project',
+        string='Project',
+        compute='_compute_related_project_id',
+        store=True,
+        states={'draft': [('readonly', False)]}
+    )
+
+    @api.model
+    def _prepare_project_vals(self, order):
+        name = "{0} - {1}".format(
+            # order.partner_id.name,
+            order.name,
+            date.today().year
+        )
+        return {
+            'user_id': order.user_id.id,
+            'name': name,
+            'partner_id': order.partner_id.id,
+        }
+
+    @api.multi
+    def action_create_project(self):
+        project_obj = self.env['project.project']
+        for order in self:
+            if order.related_project_id:
+                raise UserError(_(
+                    '''There is a project already related with this sale order.
+                        Order: {0}, Project: {1}'''.format(
+                        order,
+                        order.related_project_id
+                    )
+                ))
+            vals = self._prepare_project_vals(order)
+            project = project_obj.create(vals)
```

Comment:
why do not you use the function _create_analytic_account?

https://github.com/odoo/odoo/blob/11.0/addons/sale/models/sale.py#L574

If you use core functionality and create a project after confirm sale order, the project has name structure diferent than if you create with this module, there is not the best way.

Reply:
can you make a PR for this change?